src/nvim/testdir/test_diffmode.vim

```diff
@@ -965,6 +965,30 @@ func Test_diff_screen()
   call delete('XdiffSetup')
 endfunc
 
+func Test_diff_with_scroll_and_change()
+  CheckScreendump
```

Comment:
this test will be skipped because Nvim does not use Vim screen tests. we need a lua screen test that mimics this one. It is pretty easy, see https://github.com/neovim/neovim/blob/9edd17509fca597b847656adc8fc20fc1cc44ce5/test/functional/example_spec.lua

Reply:
Thanks for the tip. Ported this test to diff_spec.lua.